### PR TITLE
deps: update docker/cli to fix go version selection issue

### PR DESCRIPTION
**Pull request by laurazard**
see: https://github.com/docker/cli/commit/70216b662dc440faaebab531deb35f0f0c633d17

supersedes/closes https://github.com/docker/compose/pull/11271

**What I did**

bump `github.com/docker/cli` version to https://github.com/docker/cli/commit/70216b662dc440faaebab531deb35f0f0c633d17

**Related issue**

```go
⭑ make

GO111MODULE=on go build  -trimpath -tags "e2e" -ldflags "-w -X github.com/docker/compose/v2/internal.Version=v2.24.0-birthday.10-3-g785835b1.m" -o "./bin/build/docker-compose" ./cmd
# github.com/docker/cli/cli/context/store
../../../../../pkg/mod/github.com/docker/cli@v25.0.0-beta.2+incompatible/cli/context/store/storeconfig.go:6:24: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)
../../../../../pkg/mod/github.com/docker/cli@v25.0.0-beta.2+incompatible/cli/context/store/store.go:74:12: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)
../../../../../pkg/mod/github.com/docker/cli@v25.0.0-beta.2+incompatible/cli/context/store/store.go:75:23: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)
../../../../../pkg/mod/github.com/docker/cli@v25.0.0-beta.2+incompatible/cli/context/store/metadatastore.go:43:58: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)
../../../../../pkg/mod/github.com/docker/cli@v25.0.0-beta.2+incompatible/cli/context/store/metadatastore.go:48:22: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)
../../../../../pkg/mod/github.com/docker/cli@v25.0.0-beta.2+incompatible/cli/context/store/metadatastore.go:80:30: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)
# github.com/docker/cli/templates
../../../../../pkg/mod/github.com/docker/cli@v25.0.0-beta.2+incompatible/templates/templates.go:13:17: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)
# github.com/docker/cli/cli/compose/template
../../../../../pkg/mod/github.com/docker/cli@v25.0.0-beta.2+incompatible/cli/compose/template/template.go:98:45: predeclared any requires go1.18 or later (-lang was set to go1.16; check go.mod)

(and more!)
```

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**

Screenshot 2023-12-18 at 17 18 12

### CODE REVIEW

**Review by glours — APPROVED** [review]
LGTM

**Review by ndeloof — APPROVED** [review]
LGTM
mocks will need to be updated as well

**Comment by ndeloof** [comment]
supersede https://github.com/docker/compose/pull/11271

**Comment by laurazard** [comment]
> mocks will need to be updated as well

Thanks! Just pushed with updated mocks.

**Review by thaJeztah — APPROVED** [review]
LGTM

we had similar changes on the daemon side as well, but I can update my draft PR to verify those. (if all works out, also will do a new beta)

File: internal/tracing/wrap.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"context"
 
 	"go.opentelemetry.io/otel/codes"
-	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.19.0"
```
Comment (laurazard) [inline internal/tracing/wrap.go:23]: Ah OTEL semconv imports my old friend 🥲

Comment (thaJeztah) [reply internal/tracing/wrap.go:23]: 😩😫😖😣

**Comment by laurazard** [comment]
OTEL things are breaking and I've a headache, so open to ideas 🤦‍♀️ . I could make use of https://github.com/moby/buildkit/pull/4472, but that's only on the v0.12 branch, and we'd have to downgrade Buildkit to v0.11.2 which implies more changes. @milas wdyt?

**Comment by laurazard** [comment]
For example's sake, that would look like this: https://github.com/docker/compose/pull/11283/commits/7fd7d924563fb7a272c4fc50097c3d12df00f336

**Comment by thaJeztah** [comment]
> OTEL things are breaking and I've a headache, so open to ideas 🤦‍♀️ . I could make use of https://github.com/moby/buildkit/pull/4472, but that's only on the v0.12 branch


DOH! Didn't realise / forgot that was only in the v0.12 branch; we should forward-port that to master.

**Comment by thaJeztah** [comment]
- @laurazard I opened https://github.com/moby/buildkit/pull/4496

**Comment by thaJeztah** [comment]
I guess updating BuildKit to current master could also work; it's already updated to OTEL 0.45 now; https://github.com/moby/buildkit/blob/bbb6aaccb9aeb856e450c4eb535034507b074254/go.mod#L74-L88

**Comment by laurazard** [comment]
> I guess updating BuildKit to current master could also work; it's already updated to OTEL 0.45 now;

Made https://github.com/docker/buildx/pull/2170 to be able to build Compose with the latest from Buildx and Buildkit, so now this works! (without downgrading anything)

**Comment by laurazard** [comment]
Ah it still had automerge enabled 😅 Please make sure it looks good, ended up just upgrading Buildkit and Buildx dependencies.

**Pull request merged**